bin/sozo/src/ops/auth.rs

```diff
@@ -16,25 +21,55 @@ pub async fn execute(command: AuthCommand, env_metadata: Option<Environment>) ->
                 let world = WorldContract::new(world_address, &account);
+                let world_reader = WorldContractReader::new(world_address, &provider)
+                    .with_block(BlockId::Tag(BlockTag::Pending));
 
                 let mut calls = Vec::new();
 
                 for mc in models_contracts {
-                    let contract = get_contract_address(&world, mc.contract).await?;
-                    calls.push(world.grant_writer_getcall(&mc.model, &contract.into()));
+                    let model_name = parse_cairo_short_string(&mc.model)?;
+                    let model = world_reader.model_reader(&model_name).await?;
+                    let model_address = model.contract_address();
+
+                    if model_address == FieldElement::ZERO {
+                        println!(
+                            "{} Unknown model {} => {}",
+                            "[WARN]".yellow(),
+                            model_name.blue(),
+                            "IGNORED".yellow()
+                        );
+                    } else {
+                        let contract = get_contract_address(&world, mc.contract).await?;
+
+                        if model_address != contract {
+                            println!(
+                                "{} Wrong address for model {} => {}",
+                                "[WARN]".yellow(),
+                                model_name.blue(),
+                                "IGNORED".yellow()
+                            );
+                        } else {
+                            calls.push(world.grant_writer_getcall(&mc.model, &contract.into()));
+                        }
+                    }
                 }
 
-                let res = account
-                    .execute(calls)
-                    .send()
-                    .await
-                    .with_context(|| "Failed to send transaction")?;
+                if !calls.is_empty() {
+                    let res = account
+                        .execute(calls)
+                        .send()
+                        .await
+                        .with_context(|| "Failed to send transaction")?;
 
-                if transaction.wait {
-                    let receipt = TransactionWaiter::new(res.transaction_hash, &provider).await?;
-                    println!("{}", serde_json::to_string_pretty(&receipt)?);
+                    if transaction.wait {
+                        let receipt =
+                            TransactionWaiter::new(res.transaction_hash, &provider).await?;
+                        println!("{}", serde_json::to_string_pretty(&receipt)?);
+                    } else {
+                        println!("Transaction hash: {:#x}", res.transaction_hash);
+                    }
                 } else {
-                    println!("Transaction hash: {:#x}", res.transaction_hash);
+                    println!("No change to apply...");
```

Comment:
We already print a warning if the model is not included in the multicall, we may skip this one, WDYT?

Reply:
Yes, ok !